content/blog/debugging-dde.md

```diff
@@ -0,0 +1,149 @@
+---
+author: "Wes Hinsley"
+date: 2019-04-29
+title: Debugging at the edge of reason
+best: false
+---
+
+What do you do if this code...
+```
+printf("Before loop\n");
+while ((a>0) && (b>0)) {
+  printf("Inside loop\n");
+  // Other loop code
+}
+if (a>0) printf("a>0\n");
+if (b>0) printf("b>0\n");
+if ((a>0) && (b>0)) printf("Both a>0 and b>0!");
+
+```
+...produces this output?
+```
+Before loop
+a>0
+b>0
+Both a>0 and b>0!
+```
+
+This, simplified, was a situation we were in when 
+[DDE](https://github.com/mrc-ide/dde) 0.9.0 was compiled, but only on Windows,
+and only when compiled with 32-bit gcc. With 64-bits, or linux, the loop
+behaved itself perfectly.
+
+But this is almost the end of the story. It began with a failing appveyor test
+on a trivial line of code. [DDE](https://github.com/mrc-ide/dde) is a package 
+that *solves ordinary differential equations (ODEs), 
+delay differential equations (DDEs) and discrete-time difference (or recursion) 
+equations, perhaps involving delays* - and all its tests and worked fine,
```

Comment:
> and all its tests and worked fine, except one

The second "and" here is a typo?

Reply:
Well spotted - I think it's gone in the last batch of edits